### PR TITLE
Set HTTP server type and options via `config.json`

### PR DESCRIPTION
**Pull request by StpMax**
## Description

- env var `MINDSDB_HTTP_SERVER_TYPE` is now obsolete (but still can be used). `MINDSDB_HTTP_SERVER_TYPE` should be used instead.
- MINDSDB_HTTP_SERVER_TYPE can be: waitress (default), flask or gunicorn.
- as alternative, server type can be set in `config['api']['http']['server']['type']`
- HTTP server is now configurable. Relevant options can be set in `config['api']['http']['server']['config']`. Default options are:
 ```js
//  waitres 
{
    "threads": 16,
    "max_request_body_size": (1 << 30) * 10,    # 10GB
    "inbuf_overflow": (1 << 30) * 10
}

// flask
{}

// gunicorn
{
    'workers': min(multiprocessing.cpu_count(), 4),
    'timeout': 600,
    'reuse_port': True,
    'preload_app': True,
    'threads': 4
}
 ```
Any options that are supported by the server can be set there. Except host/port, they are configured in config['api']['http']['host'] and config['api']['http']['port'] as before

Fixes #issue_number

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/utilities/config.py [hunk 1]
```diff
@@ -306,6 +320,35 @@ def prepare_env_config(self) -> None:
             self._env_config['logging']['handlers']['file']['enabled'] = True
         # endregion
 
+        # region server type
+        server_type = os.environ.get('MINDSDB_HTTP_SERVER_TYPE', '').lower()
```
Comment (ZoranPandovski) [inline mindsdb/utilities/config.py:324]: Should we catch scenarios where server type is invalid, outside of supported values?

File: mindsdb/utilities/config.py [hunk 2]
```diff
@@ -443,6 +486,12 @@ def raise_warnings(self, logger) -> None:
         if 'log' in self._config:
             logger.warning("The 'log' config option is no longer supported. Use 'logging' instead.")
 
+        if os.environ.get('MINDSDB_HTTP_SERVER_TYPE', '') != '':
+            logger.warning(
+                "Env variable 'MINDSDB_HTTP_SERVER_TYPE' is going to be deprecated soon. "
```
Comment (ZoranPandovski) [inline mindsdb/utilities/config.py:491]: Isn't variable name the same here? MINDSDB_HTTP_SERVER_TYPE

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/utilities/config.py (hunk 2 above)
Comment (StpMax) [reply mindsdb/utilities/config.py:491]: indeed, fixed

File: mindsdb/utilities/config.py (hunk 1 above)
Comment (StpMax) [reply mindsdb/utilities/config.py:324]: added to `raise_warnings`

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**